composer/callbacks/memory_monitor.py

```diff
@@ -76,12 +76,11 @@ class MemoryMonitor(Callback):
         Memory usage monitoring is only supported for the GPU devices.
     """
 
-    def __init__(self):
-        super().__init__()
-        log.info(
-            "Memory monitor just profiles the current GPU assuming that the memory footprint across GPUs is balanced.")
-        if torch.cuda.device_count() == 0:
-            log.warning("Memory monitor only works on GPU devices.")
+    def init(self, state: State, logger: Logger) -> None:
+        model_device = next(state.model.parameters()).device
+
+        if model_device.type != 'cuda':
+            raise RuntimeError("The memory monitor only works on CUDA devices, but the model is on CPU.")
```

Comment:
I think the motivation to keeping this as a warning is that often users will switch between CPU and GPU for dev work, and don't want to be editing their code to add/remove this as needed. what do you think?

Reply:
True, though part of me wants invalid configurations to be an error. By dev work, would this be external users using composer or devs contributing to composer? If the former, then agree it should be a warning; if the latter, then I think it should be an error, since external ux >> developer ux.

Reply:
Seems like the popular opinion is to leave it as a warning :)